### ISSUE TITLE
Some problems with calling context menu

### ISSUE THREAD

**Issue by xmoer** [dsafa/CSDeskBand#37]
Thank the author for making this project, but I found some small problems in use.

1. Right-click on taskbar tool image is invalid. Context menu can only be displayed by right-clicking in the blank.
2. In the white taskbar of Windows10 (19H1), CSDeskBand tools will have black jagged edges.

**Comment by dsafa** [comment]
For 1. is it with winforms or wpf? do you have an example screenshot or video.
For 2. I'll take a look when the update comes out. wpf doesn't combine too well with the taskbar so there will probably be issues.
Thanks.

**Comment by dsafa** [comment]
Thanks for the visual. I believe its because wpf is not bubbling the event up to winforms. I'll see what I can do with that.

**Comment by dsafa** [comment]
I found that its because the form doesn't pass the click even through to the taskbar. I have a fix for non transparent wpf controls but for transparency i'm not currently sure on how to fix it.

**Comment by xmoer** [comment]
Yes, it's not easy to solve these two problems. I'm also studying how to fix the second problem (edge jagged for non-black background).

**Comment by xmoer** [comment]
I found a few strange problems. Control edges have color residues because TranparencyKey & BackColor is imperfect. But transparent methods do not show the same effect in various versions of Windows10.

The background of the previous system taskbar before 19H1 is black by default, so you won't see black serrated teeth. Now Microsoft has added the light mode, and the taskbar and start menu background can turn white. The problem of edge color residue became serious and I tried many methods.

First, I changed the transparent color to magenta or white, and the result was still terrible. Then I quoted the project: https://github.com/patbec/TaskbarSampleExtension. I modified the file to try to load the WPF control directly.

```
[TaskbarSampleExt/DeskBand.cs]
public class DeskBand : UserControl, IObjectWithSite, IDeskBand2 { …… }
Revised to:
public class DeskBand : ElementHost, IObjectWithSite, IDeskBand2 { …… }
```

Without specifying the transparent color, only "BackColorTransparent = true" is needed to achieve the perfect display effect. Of course, this method only works under 19H1. Of course, this method only works under 19H1, and will turn into a black background in the old system. The result surprised me.

**Comment by dsafa** [comment]
Thanks for the news. This is really interesting. I guess we'll just have to rely on microsoft's changes.

**Comment by dsafa** [comment]
In version 3.1 I updated so transparency works properly. Context menu is still a bit of an issue though

**Comment by xmoer** [comment]
> In version 3.1 I updated so transparency works properly. Context menu is still a bit of an issue though

Version 3.1 seems to have changed a lot. How do I use "INotifyPropertyChanged"?  How does "Deskband()" access wpf control properties in "UserControl1()"?

**Comment by xmoer** [comment]
Thank you. I solved the problem myself. It can be solved by following operations.
```
public partial class UserControl1 : UserControl
{
        public static UserControl1 usercontrol1;
        public UserControl1()
        {
            usercontrol1 = this;
        }
}

public class Deskband : CSDeskBandWpf, INotifyPropertyChanged
{
        public Deskband()
        {
            UserControl1.usercontrol1.Button.Visibility = Visibility.Hidden;
            ……
        }
}
```

**Comment by xmoer** [comment]
ContextMenu may not be a problem? Although it's not perfect, we can customize a menu. For example:
```
<Grid>
    <Grid.ContextMenu>
        <ContextMenu>
            <MenuItem Name="MenuItem1" Header="AAAAA"/>
            <MenuItem Name="MenuItem2" Header="BBBBB"/>
        </ContextMenu>
    </Grid.ContextMenu>
    ……
</Grid>
```

**Comment by dsafa** [comment]
Yeah, creating your own context menu is what I would recommend.
